### PR TITLE
Refresh after adding funds

### PR DESCRIPTION
**Pull request by JoyLubega**
Resolve https://github.com/opencollective/opencollective/issues/3851

# Description

This PR adds functionality for refreshing transactions in the budget section

#### loom video: https://www.loom.com/share/873016b445e14cc591b53f42c7775290

### CODE REVIEW

**Review by znarf — CHANGES_REQUESTED** [review]
I haven't tested but.

- Fix lint
- Remove edit to package-lock.json
- Make sure to pass `API_V2_CONTEXT` to `budgetSectionQuery`

**Comment by JoyLubega** [comment]
Hey @znarf  I added a PR for the api https://github.com/opencollective/opencollective-api/pull/5187 basing on the specs added. I have also made some changes here. thanks

File: components/host-dashboard/AddFundsModal.js [hunk 1]
```diff
@@ -91,7 +94,18 @@ const buildAccountReference = input => {
...
-  const [submitAddFunds, { error }] = useMutation(addFundsMutation, { context: API_V2_CONTEXT });
+  const [submitAddFunds, { error }] = useMutation(addFundsMutation, {
+    context: API_V2_CONTEXT,
+    refetchQueries: [
+      {
+        query: budgetSectionQuery,
+        context: API_V2_CONTEXT,
+        variables: getBudgetSectionQueryVariables(collective.slug),
+      },
+      { query: collectivePageQuery, variables: { slug: collective.slug } },
```
Comment (znarf) [inline components/host-dashboard/AddFundsModal.js:105]: Would be great to also use `getCollectivePageQueryVariables` here.

File: components/host-dashboard/AddFundsModal.js [hunk 2]
```diff
@@ -91,7 +94,18 @@ const buildAccountReference = input => {
...
 const AddFundsModal = ({ host, collective, ...props }) => {
   const { LoggedInUser } = useUser();
-  const [submitAddFunds, { error }] = useMutation(addFundsMutation, { context: API_V2_CONTEXT });
+  const [submitAddFunds, { error }] = useMutation(addFundsMutation, {
+    context: API_V2_CONTEXT,
+    refetchQueries: [
+      {
+        query: budgetSectionQuery,
+        context: API_V2_CONTEXT,
+        variables: getBudgetSectionQueryVariables(collective.slug),
```
Comment (znarf) [inline components/host-dashboard/AddFundsModal.js:103]: Nice!

**Review by znarf — CHANGES_REQUESTED** [review]
Looks good! Just one last thing to address.

**Comment by JoyLubega** [comment]
> Looks good! Just one last thing to address.

Thanks. Let me add `getCollectivePageQueryVariables`  now

**Review by znarf — CHANGES_REQUESTED** [review]
package-lock.json should not be changed

**Comment by JoyLubega** [comment]
> package-lock.json should not be changed

yeahhh fixed it. I rebased and it showed changes. I have reverted

**Comment by JoyLubega** [comment]
@znarf I have made the changes

**Review by znarf — APPROVED** [review]
@Betree good for you too?

**Review by Betree — COMMENTED** [review]
Yep that looks good!

**Pull request merged**